### PR TITLE
Add table aws_redshift_snapshot. closes #205

### LINKED ISSUE #205: Add table aws_redshift_snapshot
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/redshift

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_redshift_snapshot []

PRETEST: tests/aws_redshift_snapshot

TEST: tests/aws_redshift_snapshot
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_vpc.my_vpc: Creating...
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Creation complete after 14s [id=vpc-051ebdb3eb906cbdf]
aws_internet_gateway.igw: Creating...
aws_internet_gateway.igw: Creation complete after 7s [id=igw-09fdb85f85d216ea1]
aws_subnet.my_subnet2: Creating...
aws_subnet.my_subnet1: Creating...
aws_subnet.my_subnet2: Creation complete after 4s [id=subnet-07fc3921d6ef9e69a]
aws_subnet.my_subnet1: Creation complete after 4s [id=subnet-0d441c8e31cd5b4a6]
aws_redshift_subnet_group.my_subnet_group: Creating...
aws_redshift_subnet_group.my_subnet_group: Creation complete after 3s [id=turbottest87178]
aws_redshift_cluster.cluster: Creating...
aws_redshift_cluster.cluster: Still creating... [10s elapsed]
aws_redshift_cluster.cluster: Still creating... [20s elapsed]
aws_redshift_cluster.cluster: Still creating... [30s elapsed]
aws_redshift_cluster.cluster: Still creating... [40s elapsed]
aws_redshift_cluster.cluster: Still creating... [50s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m0s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m10s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m20s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m30s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m40s elapsed]
aws_redshift_cluster.cluster: Still creating... [1m50s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m0s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m10s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m21s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m31s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m41s elapsed]
aws_redshift_cluster.cluster: Still creating... [2m51s elapsed]
aws_redshift_cluster.cluster: Still creating... [3m1s elapsed]
aws_redshift_cluster.cluster: Creation complete after 3m3s [id=turbottest87178]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "      aws redshift create-cluster-snapshot --cluster-identifier turbottest87178 --snapshot-identifier turbottest87178 --tags Key=name,Value=turbottest87178 --profile default --region us-east-1;\n      # aws redshift create-tags --resource-name turbottest87178 --region us-east-1 --tags key=name,value=turbottest87178 --profile default;\n"]
null_resource.named_test_resource (local-exec): {
null_resource.named_test_resource (local-exec):     "Snapshot": {
null_resource.named_test_resource (local-exec):         "SnapshotIdentifier": "turbottest87178",
null_resource.named_test_resource (local-exec):         "ClusterIdentifier": "turbottest87178",
null_resource.named_test_resource (local-exec):         "SnapshotCreateTime": "2021-04-06T07:35:56.595000+00:00",
null_resource.named_test_resource (local-exec):         "Status": "creating",
null_resource.named_test_resource (local-exec):         "Port": 5439,
null_resource.named_test_resource (local-exec):         "AvailabilityZone": "us-east-1a",
null_resource.named_test_resource (local-exec):         "ClusterCreateTime": "2021-04-06T07:35:40.012000+00:00",
null_resource.named_test_resource (local-exec):         "MasterUsername": "rahul",
null_resource.named_test_resource (local-exec):         "ClusterVersion": "1.0",
null_resource.named_test_resource (local-exec):         "SnapshotType": "manual",
null_resource.named_test_resource (local-exec):         "NodeType": "dc2.large",
null_resource.named_test_resource (local-exec):         "NumberOfNodes": 1,
null_resource.named_test_resource (local-exec):         "DBName": "testdb",
null_resource.named_test_resource (local-exec):         "VpcId": "vpc-051ebdb3eb906cbdf",
null_resource.named_test_resource (local-exec):         "Encrypted": false,
null_resource.named_test_resource (local-exec):         "EncryptedWithHSM": false,
null_resource.named_test_resource (local-exec):         "OwnerAccount": "986325076436",
null_resource.named_test_resource (local-exec):         "TotalBackupSizeInMegaBytes": -1.0,
null_resource.named_test_resource (local-exec):         "ActualIncrementalBackupSizeInMegaBytes": -1.0,
null_resource.named_test_resource (local-exec):         "BackupProgressInMegaBytes": 0.0,
null_resource.named_test_resource (local-exec):         "CurrentBackupRateInMegaBytesPerSecond": 0.0,
null_resource.named_test_resource (local-exec):         "EstimatedSecondsToCompletion": -1,
null_resource.named_test_resource (local-exec):         "ElapsedTimeInSeconds": 0,
null_resource.named_test_resource (local-exec):         "Tags": [
null_resource.named_test_resource (local-exec):             {
null_resource.named_test_resource (local-exec):                 "Key": "name",
null_resource.named_test_resource (local-exec):                 "Value": "turbottest87178"
null_resource.named_test_resource (local-exec):             }
null_resource.named_test_resource (local-exec):         ],
null_resource.named_test_resource (local-exec):         "EnhancedVpcRouting": false,
null_resource.named_test_resource (local-exec):         "MaintenanceTrackName": "current",
null_resource.named_test_resource (local-exec):         "ManualSnapshotRetentionPeriod": -1
null_resource.named_test_resource (local-exec):     }
null_resource.named_test_resource (local-exec): }
null_resource.named_test_resource: Creation complete after 3s [id=5951316051574524596]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Warning: Quoted references are deprecated

  on variables.tf line 68, in resource "aws_subnet" "my_subnet1":
  68:       "aws_internet_gateway.igw"

In this context, references are expected literally rather than in quotes.
Terraform 0.11 and earlier required quotes, but quoted references are now
deprecated and will be removed in a future version of Terraform. Remove the
quotes surrounding this reference to silence this warning.

(and one more similar warning elsewhere)


Apply complete! Resources: 7 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:redshift:us-east-1:986325076436:snapshot:turbottest87178/turbottest87178
resource_id = 5951316051574524596
resource_name = turbottest87178

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:986325076436:snapshot:turbottest87178/turbottest87178"
    ],
    "title": "turbottest87178"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:986325076436:snapshot:turbottest87178/turbottest87178"
    ],
    "snapshot_identifier": "turbottest87178",
    "tags": {
      "name": "turbottest87178"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "cluster_identifier": "turbottest87178",
    "snapshot_identifier": "turbottest87178"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:redshift:us-east-1:986325076436:snapshot:turbottest87178/turbottest87178"
    ],
    "region": "us-east-1",
    "tags": {
      "name": "turbottest87178"
    },
    "title": "turbottest87178"
  }
]
✔ PASSED

POSTTEST: tests/aws_redshift_snapshot
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
null_resource.delay: Creating...
null_resource.delay: Provisioning with 'local-exec'...
null_resource.delay (local-exec): Executing: ["/bin/sh" "-c" "sleep 300"]
null_resource.delay: Still creating... [10s elapsed]
null_resource.delay: Still creating... [20s elapsed]
null_resource.delay: Still creating... [30s elapsed]
null_resource.delay: Still creating... [40s elapsed]
null_resource.delay: Still creating... [50s elapsed]
null_resource.delay: Still creating... [1m0s elapsed]
null_resource.delay: Still creating... [1m10s elapsed]
null_resource.delay: Still creating... [1m20s elapsed]
null_resource.delay: Still creating... [1m30s elapsed]
null_resource.delay: Still creating... [1m40s elapsed]
null_resource.delay: Still creating... [1m50s elapsed]
null_resource.delay: Still creating... [2m0s elapsed]
null_resource.delay: Still creating... [2m10s elapsed]
null_resource.delay: Still creating... [2m20s elapsed]
null_resource.delay: Still creating... [2m30s elapsed]
null_resource.delay: Still creating... [2m40s elapsed]
null_resource.delay: Still creating... [2m50s elapsed]
null_resource.delay: Still creating... [3m0s elapsed]
null_resource.delay: Still creating... [3m10s elapsed]
null_resource.delay: Still creating... [3m20s elapsed]
null_resource.delay: Still creating... [3m30s elapsed]
null_resource.delay: Still creating... [3m40s elapsed]
null_resource.delay: Still creating... [3m50s elapsed]
null_resource.delay: Still creating... [4m0s elapsed]
null_resource.delay: Still creating... [4m10s elapsed]
null_resource.delay: Still creating... [4m20s elapsed]
null_resource.delay: Still creating... [4m30s elapsed]
null_resource.delay: Still creating... [4m40s elapsed]
null_resource.delay: Still creating... [4m50s elapsed]
null_resource.delay: Creation complete after 5m0s [id=6136276321967741642]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "      aws redshift delete-cluster-snapshot --snapshot-identifier turbottest87178 --profile default --region us-east-1;\n      # aws redshift delete-tags --resource-name turbottest87178 --tag-keys=name --region us-east-1  --profile default;\n"]
null_resource.named_test_resource (local-exec): {
null_resource.named_test_resource (local-exec):     "Snapshot": {
null_resource.named_test_resource (local-exec):         "SnapshotIdentifier": "turbottest87178",
null_resource.named_test_resource (local-exec):         "ClusterIdentifier": "turbottest87178",
null_resource.named_test_resource (local-exec):         "SnapshotCreateTime": "2021-04-06T07:39:16.408000+00:00",
null_resource.named_test_resource (local-exec):         "Status": "deleted",
null_resource.named_test_resource (local-exec):         "Port": 5439,
null_resource.named_test_resource (local-exec):         "AvailabilityZone": "us-east-1a",
null_resource.named_test_resource (local-exec):         "ClusterCreateTime": "2021-04-06T07:35:40.012000+00:00",
null_resource.named_test_resource (local-exec):         "MasterUsername": "rahul",
null_resource.named_test_resource (local-exec):         "ClusterVersion": "1.0",
null_resource.named_test_resource (local-exec):         "SnapshotType": "manual",
null_resource.named_test_resource (local-exec):         "NodeType": "dc2.large",
null_resource.named_test_resource (local-exec):         "NumberOfNodes": 1,
null_resource.named_test_resource (local-exec):         "DBName": "testdb",
null_resource.named_test_resource (local-exec):         "VpcId": "vpc-051ebdb3eb906cbdf",
null_resource.named_test_resource (local-exec):         "Encrypted": false,
null_resource.named_test_resource (local-exec):         "EncryptedWithHSM": false,
null_resource.named_test_resource (local-exec):         "OwnerAccount": "986325076436",
null_resource.named_test_resource (local-exec):         "TotalBackupSizeInMegaBytes": 26.0,
null_resource.named_test_resource (local-exec):         "ActualIncrementalBackupSizeInMegaBytes": 9.0,
null_resource.named_test_resource (local-exec):         "BackupProgressInMegaBytes": 9.0,
null_resource.named_test_resource (local-exec):         "CurrentBackupRateInMegaBytesPerSecond": 8.1081,
null_resource.named_test_resource (local-exec):         "EstimatedSecondsToCompletion": 0,
null_resource.named_test_resource (local-exec):         "ElapsedTimeInSeconds": 1,
null_resource.named_test_resource (local-exec):         "Tags": [],
null_resource.named_test_resource (local-exec):         "EnhancedVpcRouting": false,
null_resource.named_test_resource (local-exec):         "MaintenanceTrackName": "current",
null_resource.named_test_resource (local-exec):         "ManualSnapshotRetentionPeriod": -1,
null_resource.named_test_resource (local-exec):         "SnapshotRetentionStartTime": "2021-04-06T07:39:17.518000+00:00"
null_resource.named_test_resource (local-exec):     }
null_resource.named_test_resource (local-exec): }
null_resource.named_test_resource: Creation complete after 3s [id=5088102859401756508]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

TEARDOWN: tests/aws_redshift_snapshot


SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
## Redshift snapshot basic info

```sql
select
  snapshot_identifier,
  node_type,
  encrypted
from
  aws_redshift_snapshot;
```
```
+---------------------+--------------------+-----------+
| snapshot_identifier | cluster_identifier | encrypted |
+---------------------+--------------------+-----------+
| turbottest29264     | turbottest29264    | false     |
| turbottest82076     | turbottest82076    | false     |
| turbottest44087     | turbottest44087    | false     |
| turbottest56460     | turbottest56460    | false     |
| turbottest68110     | turbottest68110    | false     |
| turbottest80541     | turbottest80541    | false     |
| turbottest1135      | turbottest1135     | false     |
| turbottest2736      | turbottest2736     | false     |
| turbottest16235     | turbottest16235    | false     |
| turbottest79003     | turbottest79003    | false     |
| turbottest73279     | turbottest73279    | false     |
| turbottest14536     | turbottest14536    | false     |
| turbottest8100      | turbottest8100     | false     |
| turbottest22845     | turbottest22845    | false     |
+---------------------+--------------------+-----------+
```

### List of all manual redshift snapshots

```sql
select
  snapshot_identifier,
  snapshot_type
from
  aws_redshift_snapshot
where
  snapshot_type = 'manual';
```
```
+---------------------+---------------+
| snapshot_identifier | snapshot_type |
+---------------------+---------------+
| turbottest14536     | manual        |
| turbottest68110     | manual        |
| turbottest80541     | manual        |
| turbottest2736      | manual        |
| turbottest16235     | manual        |
| turbottest8100      | manual        |
| turbottest56460     | manual        |
| turbottest82076     | manual        |
| turbottest79003     | manual        |
| turbottest29264     | manual        |
| turbottest44087     | manual        |
| turbottest22845     | manual        |
| turbottest1135      | manual        |
| turbottest73279     | manual        |
+---------------------+---------------+
```

### List of snapshots which are not encrypted

```sql
select
  snapshot_identifier,
  cluster_identifier,
  node_type,
  number_of_nodes,
  encrypted
from
  aws_redshift_snapshot
where
  not encrypted;
```
```
+---------------------+--------------------+-----------+-----------------+-----------+
| snapshot_identifier | cluster_identifier | node_type | number_of_nodes | encrypted |
+---------------------+--------------------+-----------+-----------------+-----------+
| turbottest14536     | turbottest14536    | dc2.large | 1               | false     |
| turbottest73279     | turbottest73279    | dc2.large | 1               | false     |
| turbottest68110     | turbottest68110    | dc2.large | 1               | false     |
| turbottest79003     | turbottest79003    | dc2.large | 1               | false     |
| turbottest80541     | turbottest80541    | dc2.large | 1               | false     |
| turbottest1135      | turbottest1135     | dc2.large | 1               | false     |
| turbottest22845     | turbottest22845    | dc2.large | 1               | false     |
| turbottest2736      | turbottest2736     | dc2.large | 1               | false     |
| turbottest82076     | turbottest82076    | dc2.large | 1               | false     |
| turbottest16235     | turbottest16235    | dc2.large | 1               | false     |
| turbottest44087     | turbottest44087    | dc2.large | 1               | false     |
| turbottest29264     | turbottest29264    | dc2.large | 1               | false     |
| turbottest8100      | turbottest8100     | dc2.large | 1               | false     |
| turbottest56460     | turbottest56460    | dc2.large | 1               | false     |
+---------------------+--------------------+-----------+-----------------+-----------+
```

### Redshift Cluster info of each redshift snapshot

```sql
select
  snapshot_identifier,
  cluster_identifier,
  number_of_nodes,
  cluster_version,
  engine_full_version,
  restorable_node_types
from
  aws_redshift_snapshot;
```
```
+---------------------+--------------------+-----------------+-----------------+---------------------+-----------------------+
| snapshot_identifier | cluster_identifier | number_of_nodes | cluster_version | engine_full_version | restorable_node_types |
+---------------------+--------------------+-----------------+-----------------+---------------------+-----------------------+
| turbottest68110     | turbottest68110    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest29264     | turbottest29264    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest1135      | turbottest1135     | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest14536     | turbottest14536    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest82076     | turbottest82076    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest16235     | turbottest16235    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest8100      | turbottest8100     | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest73279     | turbottest73279    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest2736      | turbottest2736     | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest22845     | turbottest22845    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest79003     | turbottest79003    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest56460     | turbottest56460    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest44087     | turbottest44087    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
| turbottest80541     | turbottest80541    | 1               | 1.0             | 1.0.24421           | ["dc2.large"]         |
+---------------------+--------------------+-----------------+-----------------+---------------------+-----------------------+
```

### List snapshots that are shared with other account

```sql
select
  snapshot_identifier,
  accounts_with_restore_access
from
  aws_redshift_snapshot
where
  accounts_with_restore_access is not null;
```
```
+-------------------------------------+-----------------------------------------------------------------------------------------------------+
| snapshot_identifier                 | accounts_with_restore_access                                                                        |
+-------------------------------------+-----------------------------------------------------------------------------------------------------+
| redshift-cluster-1-2021-04-12-13-44 | [{"AccountAlias":null,"AccountId":"027337808343"},{"AccountAlias":null,"AccountId":"856090093175"}] |
| redshift-cluster-1-2021-04-06-17-17 | [{"AccountAlias":null,"AccountId":"856090093175"}]                                                  |
+-------------------------------------+-----------------------------------------------------------------------------------------------------+
```

### List accounts that are authorised to restore the snapshots

```sql
select
 snapshot_identifier,
 p ->> 'AccountId' as account_id,
 p ->> 'AccountAlias' as account_alias
from
 aws_redshift_snapshot,
 jsonb_array_elements(accounts_with_restore_access) as p;
```
```
+---------------------+------------+---------------+
| snapshot_identifier | account_id | account_alias |
+---------------------+------------+---------------+
+---------------------+------------+---------------+
(Note: Since all accounts are unauthorised to restore snapshots)
+-------------------------------------------+------------------------------+
| snapshot_identifier                       | accounts_with_restore_access |
+-------------------------------------------+------------------------------+
| rs:redshift-cluster-1-2021-04-06-03-36-33 | <null>                       |
| rs:redshift-cluster-1-2021-04-05-19-36-13 | <null>                       |
+-------------------------------------------+------------------------------+
```

### List snapshots of respective clusters those are not in a VPC

```sql
select
  snapshot_identifier,
  cluster_identifier,
  node_type,
  vpc_id
from
  aws_redshift_snapshot
where
  vpc_id is null;
```
```
+---------------------+--------------------+-----------+--------+
| snapshot_identifier | cluster_identifier | node_type | vpc_id |
+---------------------+--------------------+-----------+--------+
+---------------------+--------------------+-----------+--------+
```

### CODE REVIEW

File: aws-test/tests/aws_redshift_snapshot/posttest-variables.tf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  default = ""
+  description = "Turbot credentials profile to use for the test run."
+}
+
+provider "turbot" {
+  profile = var.turbot_profile
+}
+
+
+
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_snapshot/posttest-variables.tf:18]: remove extra lines

File: aws-test/tests/aws_redshift_snapshot/test-turbot-expected.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "account_id": "{{output.account_id.value}}",
+    "akas": [
+      "{{ output.resource_aka.value }}"
+    ],
+    "region": "{{ output.aws_region.value }}",
+    "tags": "<null>",
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_snapshot/test-turbot-expected.json:8]: can you please add a tag and test for its value?

File: aws/plugin.go [hunk 3]
```diff
@@ -82,6 +82,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_lambda_alias":                       tableAwsLambdaAlias(ctx),
 			"aws_lambda_function":                    tableAwsLambdaFunction(ctx),
 			"aws_lambda_version":                     tableAwsLambdaVersion(ctx),
+			"aws_redshift_snapshot":                  tableAwsRedshiftSnapshot(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:85]: sort it

File: aws/table_aws_redshift_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+				Name:        "vpc_id",
+				Description: "The VPC identifier of the cluster if the snapshot is from a cluster in a VPC.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "tags_src",
+				Description: "The list of tags for the cluster.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
```
Comment (Paulami30) [inline aws/table_aws_redshift_snapshot.go:200]:
```suggestion
			},
			
```

File: aws/table_aws_redshift_snapshot.go [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+			{
+				Name:        "port",
+				Description: "The port that the cluster is listening on.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "restorable_node_types",
+				Description: "The list of node types that this cluster snapshot is able to restore into.",
+				Type:        proto.ColumnType_JSON,
+			},
```
Comment (Paulami30) [inline aws/table_aws_redshift_snapshot.go:159]: arrange the columns in standard column format

File: aws/table_aws_redshift_snapshot.go [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getRedshiftSnapshotAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+func listRedshiftSnapshot(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_redshift_snapshot.go:225]:
```suggestion
func listRedshiftSnapshots(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: docs/tables/aws_redshift_snapshot.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
+
+# Table: aws_redshift_snapshot
+
+Snapshots are point-in-time backups of a cluster. There are two types of snapshots: automated and manual. Amazon Redshift stores these snapshots internally in Amazon S3 by using an encrypted Secure Sockets Layer (SSL) connection.
+
+## Examples
+
+### Redshift snapshot basic info
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:8]:
```suggestion
### Basic info
```

File: docs/tables/aws_redshift_snapshot.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  snapshot_identifier,
+  cluster_identifier,
+  node_type,
+  encrypted
+from
+  aws_redshift_snapshot;
+```
+
+
+### List of all manual redshift snapshots
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:21]:
```suggestion
### List of manual redshift snapshots
```

File: docs/tables/aws_redshift_snapshot.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+  snapshot_identifier,
+  snapshot_type
+from
+  aws_redshift_snapshot
+where
+  snapshot_type = 'manual';
+```
+
+
+### List of snapshots which are not encrypted
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:34]:
```suggestion
### List snapshots which are not encrypted
```

File: docs/tables/aws_redshift_snapshot.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  number_of_nodes,
+  encrypted
+from
+  aws_redshift_snapshot
+where
+  not encrypted;
+```
+
+
+### Redshift Cluster info of each redshift snapshot
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:50]:
```suggestion
### Get redshift cluster info of each redshift snapshot
```

**Review by Paulami30 — COMMENTED** [review]
Please check the comments, run the integration test and paste the current log. Thank you.

File: aws-test/tests/aws_redshift_snapshot/variables.tf [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+  vpc_id = "${aws_vpc.my_vpc.id}"
+  tags = {
+    Name = var.resource_name
+  }
+}
+
+resource "aws_subnet" "my_subnet1" {
+  cidr_block        = "10.1.1.0/24"
+  availability_zone = "${var.aws_region}a"
+  vpc_id            = "${aws_vpc.my_vpc.id}"
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_snapshot/variables.tf:66]:
```suggestion
  vpc_id            = aws_vpc.my_vpc.id
```

File: aws-test/tests/aws_redshift_snapshot/variables.tf [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    ]
+  tags = {
+    Name = var.resource_name
+  }
+}
+
+resource "aws_subnet" "my_subnet2" {
+  cidr_block        = "10.1.2.0/24"
+  availability_zone = "${var.aws_region}b"
+  vpc_id            = "${aws_vpc.my_vpc.id}"
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_snapshot/variables.tf:78]:
```suggestion
  vpc_id            = aws_vpc.my_vpc.id
```

File: aws-test/tests/aws_redshift_snapshot/variables.tf [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+resource "aws_vpc" "my_vpc" {
+  cidr_block = "10.1.0.0/16"
+  tags = {
+    Name = var.resource_name
+  }
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = "${aws_vpc.my_vpc.id}"
```
Comment (Paulami30) [inline aws-test/tests/aws_redshift_snapshot/variables.tf:57]:
```suggestion
  vpc_id = aws_vpc.my_vpc.id
```

File: aws-test/tests/aws_redshift_snapshot/test-turbot-expected.json (hunk 2 above)
Comment (Paulami30) [reply aws-test/tests/aws_redshift_snapshot/test-turbot-expected.json:8]: this change is not done yet

**Review by Paulami30 — COMMENTED** [review]
Please check all the comments and also run the integration test and add the new logs

File: aws/plugin.go [hunk 14]
```diff
@@ -107,6 +107,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_rds_db_parameter_group":             tableAwsRDSDBParameterGroup(ctx),
 			"aws_rds_db_snapshot":                    tableAwsRDSDBSnapshot(ctx),
 			"aws_rds_db_subnet_group":                tableAwsRDSDBSubnetGroup(ctx),
+			"aws_redshift_snapshot":                  tableAwsRedshiftSnapshot(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:110]: sort the list

File: aws/table_aws_redshift_snapshot.go [hunk 15]
```diff
@@ -0,0 +1,320 @@
...
+
+	commonColumnData := c.(*awsCommonColumnData)
+	aka := []string{"arn:" + commonColumnData.Partition + ":redshift:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":snapshot:" + *snapshot.ClusterIdentifier + "/" + *snapshot.SnapshotIdentifier}
+
+	return aka, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getRedshiftSnapshotTurbotTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (Paulami30) [inline aws/table_aws_redshift_snapshot.go:309]:
```suggestion
func tagListToTurbotTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
```

File: docs/tables/aws_redshift_snapshot.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+  node_type,
+  vpc_id
+from
+  aws_redshift_snapshot
+where
+  vpc_id is null;
+```
+
+
+### List snapshots whose AWS customer accounts are not authorized
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:80]:
```suggestion
### List of AWS accounts authorised to restore the snapshots
```

File: docs/tables/aws_redshift_snapshot.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+```sql
+select
+  snapshot_identifier,
+  accounts_with_restore_access
+from
+  aws_redshift_snapshot
+where
+  accounts_with_restore_access is null;
+
+```
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:91]: Modify this to query the list of accounts that are authorised to restore the snapshots

File: docs/tables/aws_redshift_snapshot.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
+
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:1]:
```suggestion
```

File: docs/tables/aws_redshift_snapshot.md [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+  node_type,
+  vpc_id
+from
+  aws_redshift_snapshot
+where
+  vpc_id is null;
+```
+
+
+### list of accounts that are authorized to restore the snapshots
```
Comment (Paulami30) [inline docs/tables/aws_redshift_snapshot.md:80]:
```suggestion
### List accounts that are authorized to restore the snapshots
```

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

File: docs/tables/aws_redshift_snapshot.md (hunk 17 above)
Comment (rajmohanty17) [reply docs/tables/aws_redshift_snapshot.md:89]: can we query to see if any manual snapshot is shared with any account ? see here https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-snapshots.html#working-with-snapshot-share-snapshot

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Pls see the comment.

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by Paulami30 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**